.github/workflows/test.yml

```diff
@@ -6,7 +6,9 @@ on:
   push:
     branches: [ main ]
   pull_request:
- 
+permissions:
+  contents: read
+  checks: write
```

Comment:
Added due to seeing this CodeQL warning 

<img width="720" alt="Screenshot 2025-05-19 at 8 44 22 PM" src="https://github.com/user-attachments/assets/39fa4efc-cbc6-403d-887a-d0404eaa0448" />